apache_exporter.go

```diff
@@ -305,22 +306,31 @@ func (e *Exporter) Collect(ch chan<- prometheus.Metric) {
 }
 
 func main() {
-	flag.Parse()
+
+	// Parse flags
+	log.AddFlags(kingpin.CommandLine)
+	kingpin.Version(version.Print("apache_exporter"))
+	kingpin.HelpFlag.Short('h')
+	kingpin.Parse()
+
 	// listen to termination signals from the OS
 	signal.Notify(gracefulStop, syscall.SIGTERM)
 	signal.Notify(gracefulStop, syscall.SIGINT)
 	signal.Notify(gracefulStop, syscall.SIGHUP)
 	signal.Notify(gracefulStop, syscall.SIGQUIT)
 
-	if *showVersion {
-		fmt.Fprintln(os.Stdout, version.Print("apache_exporter"))
-		os.Exit(0)
-	}
 	exporter := NewExporter(*scrapeURI)
 	prometheus.MustRegister(exporter)
 	prometheus.MustRegister(version.NewCollector("apache_exporter"))
 
 	log.Infoln("Starting apache_exporter", version.Info())
+	log.Debugf("With flags:\n\t"+
+		"telemetry.address: %s\n\t"+
+		"telemetry.endpoint: %s\n\t"+
+		"scrape_uri: %s\n\t"+
```

Comment:
scrape_uri may contain basic auth credentials which may be stored in plain text to syslog